src/gettxinfo.c

```diff
@@ -188,12 +189,29 @@ void gettxinfo(void) {
 	    handle_trx_bandswitch((int) freq);
 	}
 
+	/* read speed from rig */
+	if (cwkeyer == HAMLIB_KEYER) {
+	    value_t rig_cwspeed;
+	    retval = rig_get_level(my_rig, RIG_VFO_CURR, RIG_LEVEL_KEYSPD, &rig_cwspeed); /* initialize RIG_VFO_CURR */
+
+	    if (retval == RIG_OK) {
+		if (GetCWSpeed() != rig_cwspeed.i) { // FIXME: doesn't work if rig speed is between the values from CW_SPEEDS
+		    SetCWSpeed(rig_cwspeed.i);
+
+		    attron(COLOR_PAIR(C_HEADER) | A_STANDOUT);
```

Comment:
Not sure if UI stuff is needed here.

Reply:
Yes, that needs to be moved elsewhere, that's the "minor bug" I was mentioning above.